# mayhem_program::instructions::burn::handler  (yours 5856 B at .text+0xe8b8; original 5752 B at .text+0xd778)
# distance 588 instructions; evidence: called-by mayhem_program::__private::__global::burn_tokens#d0c0; same-callees 13/13
--- original
+++ yours
@@ -1,296 +1,210 @@
 mov64 r7, r1
-ldxdw r8, [r7+8]
-ldxdw r6, [r8+424]
-mov64 r1, r10
-add64 r1, -336
+ldxdw r9, [r7+8]
+ldxdw r6, [r9+424]
+mov64 r1, r10
+add64 r1, -328
 call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
-ldxw r1, [r10-336]
-jeq r1, 1, +0x1508
-ldxdw r1, [r10-296]
-jsle r6, r1, +0x1a0
-mov64 r1, r10
-add64 r1, -120
-lddw r2, .rodata[05000000]
-call mayhem_program::MayhemError::name#1311
-stdw [r10-72], 0
-stdw [r10-80], 1
-stdw [r10-88], 0
-lddw r1, table[0:core::ptr::drop_in_place<alloc::string::String>#35e0, 3:<alloc::string::String as core::fmt::Write>::write_str#ca2d, 4:<alloc::string::String as core::fmt::Write>::write_char#6537, 5:core::fmt::Write::write_fmt#5fac]
+ldxw r1, [r10-328]
+jeq r1, 1, +0x15c0
+ldxdw r1, [r10-288]
+jsgt r6, r1, +0xd40
+ldxb r1, [r9+448]
+jeq r1, 2, +0xf08
+ldxdw r0, [r9+232]
+ldxb r1, [r7+33]
+stxb [r10-361], r1
+mov64 r1, r10
+add64 r1, -361
+stxdw [r10-344], r1
+lddw r1, .rodata[736f6c2d7661756c]
+stxdw [r10-360], r1
+stdw [r10-336], 1
+stdw [r10-352], 9
+ldxdw r1, [r9+504]
+ldxdw r1, [r1]
+ldxdw r2, [r1+24]
+stxdw [r10-24], r2
+ldxdw r2, [r1+16]
+stxdw [r10-32], r2
+ldxdw r2, [r1+8]
+stxdw [r10-40], r2
+ldxdw r1, [r1]
 stxdw [r10-48], r1
-mov64 r1, r10
-add64 r1, -88
-stxdw [r10-56], r1
-lddw r1, 0xe0000020
-stxdw [r10-40], r1
-mov64 r2, r10
-add64 r2, -56
-lddw r1, .rodata[05000000]
-call orig_fn_2fcb8
-jne r0, 0, +0x1568
-ldxdw r1, [r10-72]
-stxdw [r10-264], r1
-ldxdw r1, [r10-80]
-stxdw [r10-272], r1
-ldxdw r1, [r10-88]
-stxdw [r10-280], r1
-ldxdw r1, [r10-120]
-stxdw [r10-304], r1
-ldxdw r1, [r10-112]
-stxdw [r10-296], r1
-ldxdw r1, [r10-104]
-stxdw [r10-288], r1
-lddw r1, .rodata[70726f6772616d73]
-stxdw [r10-328], r1
-stw [r10-184], 6005
-stb [r10-256], 2
-stw [r10-312], 78
-ja +0x2f8
-ldxb r1, [r8+448]
-jne r1, 2, +0x328
-mov64 r1, r10
-add64 r1, -120
-lddw r2, .rodata[04000000]
-call mayhem_program::MayhemError::name#1311
-stdw [r10-72], 0
-stdw [r10-80], 1
-stdw [r10-88], 0
-lddw r1, table[0:core::ptr::drop_in_place<alloc::string::String>#35e0, 3:<alloc::string::String as core::fmt::Write>::write_str#ca2d, 4:<alloc::string::String as core::fmt::Write>::write_char#6537, 5:core::fmt::Write::write_fmt#5fac]
-stxdw [r10-48], r1
-mov64 r1, r10
-add64 r1, -88
-stxdw [r10-56], r1
-lddw r1, 0xe0000020
-stxdw [r10-40], r1
-mov64 r2, r10
-add64 r2, -56
-lddw r1, .rodata[04000000]
-call orig_fn_2fcb8
-jne r0, 0, +0x1568
-ldxdw r1, [r10-72]
-stxdw [r10-264], r1
-ldxdw r1, [r10-80]
-stxdw [r10-272], r1
-ldxdw r1, [r10-88]
-stxdw [r10-280], r1
-ldxdw r1, [r10-120]
-stxdw [r10-304], r1
-ldxdw r1, [r10-112]
-stxdw [r10-296], r1
-ldxdw r1, [r10-104]
-stxdw [r10-288], r1
-lddw r1, .rodata[70726f6772616d73]
-stxdw [r10-328], r1
-stw [r10-184], 6004
-stb [r10-256], 2
-stw [r10-312], 82
-stdw [r10-320], 48
-stdw [r10-336], 0
-mov64 r1, r10
-add64 r1, -336
-call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
-exit
-ldxdw r1, [r8+232]
-stxdw [r10-344], r1
-ldxdw r1, [r8+504]
-ldxdw r1, [r1]
-ldxdw r2, [r1+24]
-stxdw [r10-64], r2
-ldxdw r2, [r1+16]
-stxdw [r10-72], r2
-ldxdw r2, [r1+8]
-stxdw [r10-80], r2
-ldxdw r1, [r1]
-stxdw [r10-88], r1
-ldxdw r2, [r8+88]
-ldxdw r4, [r2+8]
-ldxdw r1, [r4]
+ldxdw r2, [r9+88]
+ldxdw r7, [r2+8]
+ldxdw r1, [r7]
 add64 r1, 1
 mov64 r3, 1
-jeq r1, 0, +0x3c0
+jeq r1, 0, +0x140
 mov64 r3, 0
-ldxdw r5, [r2]
-stxdw [r10-352], r5
-stxdw [r4], r1
-jeq r3, 1, +0x1500
-ldxdw r0, [r2+16]
+ldxdw r4, [r2]
+stxdw [r10-392], r4
+stxdw [r7], r1
+jeq r3, 1, +0x15b8
+ldxdw r6, [r2+16]
+ldxdw r1, [r6]
+add64 r1, 1
+mov64 r3, 1
+jeq r1, 0, +0x190
+mov64 r3, 0
+stxdw [r6], r1
+jeq r3, 1, +0x15b8
+ldxdw r5, [r9+160]
+ldxdw r8, [r5+8]
+ldxdw r1, [r8]
+add64 r1, 1
+mov64 r3, 1
+jeq r1, 0, +0x1d8
+mov64 r3, 0
+ldxb r4, [r2+42]
+stxdw [r10-424], r4
+ldxb r4, [r2+41]
+stxdw [r10-416], r4
+ldxb r4, [r2+40]
+stxdw [r10-408], r4
+ldxdw r4, [r2+32]
+stxdw [r10-400], r4
+ldxdw r2, [r2+24]
+ldxdw r4, [r5]
+stxdw [r8], r1
+jeq r3, 1, +0x15b8
+stxdw [r10-440], r2
+stxdw [r10-432], r8
+stxdw [r10-384], r0
+ldxdw r8, [r5+16]
+ldxdw r1, [r8]
+add64 r1, 1
+mov64 r2, 1
+jeq r1, 0, +0x280
+mov64 r2, 0
+stxdw [r8], r1
+jeq r2, 1, +0x15b8
+stxdw [r10-448], r6
+ldxdw r6, [r9+456]
+ldxdw r0, [r6+8]
 ldxdw r1, [r0]
 add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x410
-mov64 r3, 0
+mov64 r2, 1
+jeq r1, 0, +0x2d0
+mov64 r2, 0
+ldxb r3, [r5+42]
+stxdw [r10-480], r3
+ldxb r3, [r5+41]
+stxdw [r10-472], r3
+ldxb r3, [r5+40]
+stxdw [r10-464], r3
+ldxdw r3, [r5+32]
+stxdw [r10-456], r3
+ldxdw r3, [r5+24]
+ldxdw r5, [r6]
 stxdw [r0], r1
-jeq r3, 1, +0x1500
-ldxdw r6, [r8+160]
-ldxdw r5, [r6+8]
-ldxdw r1, [r5]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x458
-mov64 r3, 0
-ldxb r9, [r2+42]
-stxdw [r10-384], r9
-ldxb r9, [r2+41]
+jeq r2, 1, +0x15b8
+stxdw [r10-488], r5
+ldxdw r1, [r6+16]
+ldxdw r2, [r1]
+add64 r2, 1
+mov64 r5, 1
+jeq r2, 0, +0x368
+mov64 r5, 0
+stxdw [r1], r2
+jeq r5, 1, +0x15b8
 stxdw [r10-376], r9
-ldxb r9, [r2+40]
-stxdw [r10-368], r9
-ldxdw r9, [r2+32]
-stxdw [r10-360], r9
-ldxdw r2, [r2+24]
-ldxdw r9, [r6]
-stxdw [r5], r1
-jeq r3, 1, +0x1500
-stxdw [r10-400], r2
-stxdw [r10-392], r5
-ldxdw r5, [r6+16]
-ldxdw r1, [r5]
-add64 r1, 1
-mov64 r2, 1
-jeq r1, 0, +0x4f8
-mov64 r2, 0
-stxdw [r5], r1
-jeq r2, 1, +0x1500
-stxdw [r10-408], r0
-stxdw [r10-480], r8
-ldxdw r8, [r8+456]
-ldxdw r0, [r8+8]
-ldxdw r2, [r0]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x550
-mov64 r3, 0
-ldxb r1, [r6+42]
-stxdw [r10-440], r1
-ldxb r1, [r6+41]
-stxdw [r10-432], r1
-ldxb r1, [r6+40]
-stxdw [r10-424], r1
-ldxdw r1, [r6+32]
-stxdw [r10-416], r1
-ldxdw r1, [r6+24]
-ldxdw r6, [r8]
-stxdw [r0], r2
-jeq r3, 1, +0x1500
-stxdw [r10-448], r6
-ldxdw r2, [r8+16]
-ldxdw r3, [r2]
-add64 r3, 1
-mov64 r6, 1
-jeq r3, 0, +0x5e8
-mov64 r6, 0
-stxdw [r2], r3
-jeq r6, 1, +0x1500
-ldxdw r3, [r8+24]
-ldxdw r6, [r8+32]
-stxdw [r10-472], r5
-stxdw [r10-464], r9
-ldxb r9, [r8+40]
-mov64 r5, r1
-ldxb r1, [r8+41]
-ldxb r8, [r8+42]
-ldxb r7, [r7+33]
-stxdw [r10-456], r4
-mov64 r4, r10
-add64 r4, -1
-stxdw [r10-40], r4
-lddw r4, .rodata[736f6c2d7661756c]
-stxdw [r10-56], r4
-stxb [r10-1], r7
-mov64 r4, r10
-add64 r4, -56
-stxdw [r10-120], r4
-stxb [r10-174], r8
-stxb [r10-175], r1
-stxb [r10-176], r9
-stxdw [r10-184], r6
-stxdw [r10-192], r3
-stxdw [r10-200], r2
-stxdw [r10-208], r0
+ldxdw r2, [r6+24]
+ldxdw r5, [r6+32]
+stxdw [r10-496], r7
+ldxb r7, [r6+40]
+ldxb r9, [r6+41]
+ldxb r6, [r6+42]
+stxb [r10-166], r6
+stxb [r10-167], r9
+stxb [r10-168], r7
+stxdw [r10-176], r5
+stxdw [r10-184], r2
+stxdw [r10-192], r1
+stxdw [r10-200], r0
+ldxdw r1, [r10-488]
+stxdw [r10-208], r1
+ldxdw r1, [r10-480]
+stxb [r10-214], r1
+ldxdw r1, [r10-472]
+stxb [r10-215], r1
+ldxdw r1, [r10-464]
+stxb [r10-216], r1
+ldxdw r1, [r10-456]
+stxdw [r10-224], r1
+stxdw [r10-232], r3
+stxdw [r10-240], r8
+ldxdw r1, [r10-432]
+stxdw [r10-248], r1
+stxdw [r10-256], r4
+ldxdw r1, [r10-424]
+stxb [r10-262], r1
+ldxdw r1, [r10-416]
+stxb [r10-263], r1
+ldxdw r1, [r10-408]
+stxb [r10-264], r1
+ldxdw r1, [r10-400]
+stxdw [r10-272], r1
+ldxdw r1, [r10-440]
+stxdw [r10-280], r1
 ldxdw r1, [r10-448]
-stxdw [r10-216], r1
-ldxdw r1, [r10-440]
-stxb [r10-222], r1
-ldxdw r1, [r10-432]
-stxb [r10-223], r1
-ldxdw r1, [r10-424]
-stxb [r10-224], r1
-ldxdw r1, [r10-416]
-stxdw [r10-232], r1
-stxdw [r10-240], r5
-ldxdw r1, [r10-472]
-stxdw [r10-248], r1
+stxdw [r10-288], r1
+ldxdw r1, [r10-496]
+stxdw [r10-296], r1
 ldxdw r1, [r10-392]
-stxdw [r10-256], r1
-ldxdw r1, [r10-464]
-stxdw [r10-264], r1
-ldxdw r1, [r10-384]
-stxb [r10-270], r1
-ldxdw r1, [r10-376]
-stxb [r10-271], r1
-ldxdw r1, [r10-368]
-stxb [r10-272], r1
-ldxdw r1, [r10-360]
-stxdw [r10-280], r1
-ldxdw r1, [r10-400]
-stxdw [r10-288], r1
-ldxdw r1, [r10-408]
-stxdw [r10-296], r1
-ldxdw r1, [r10-456]
 stxdw [r10-304], r1
-ldxdw r1, [r10-352]
-stxdw [r10-312], r1
-stdw [r10-32], 1
-stdw [r10-48], 9
-stdw [r10-112], 2
-mov64 r1, r10
-add64 r1, -120
-stxdw [r10-168], r1
-ldxdw r1, [r10-88]
-stxdw [r10-152], r1
-ldxdw r1, [r10-80]
+mov64 r1, r10
+add64 r1, -360
+stxdw [r10-80], r1
+stdw [r10-72], 2
+mov64 r1, r10
+add64 r1, -80
+stxdw [r10-160], r1
+ldxdw r1, [r10-48]
 stxdw [r10-144], r1
-ldxdw r1, [r10-72]
+ldxdw r1, [r10-40]
 stxdw [r10-136], r1
-ldxdw r1, [r10-64]
+ldxdw r1, [r10-32]
 stxdw [r10-128], r1
-stdw [r10-320], 0
-stdw [r10-328], 8
-stdw [r10-336], 0
-stdw [r10-160], 1
-mov64 r1, r10
-add64 r1, -336
-ldxdw r8, [r10-344]
-mov64 r2, r8
+ldxdw r1, [r10-24]
+stxdw [r10-120], r1
+stdw [r10-312], 0
+stdw [r10-320], 8
+stdw [r10-328], 0
+stdw [r10-152], 1
+mov64 r1, r10
+add64 r1, -328
+ldxdw r7, [r10-384]
+mov64 r2, r7
 call anchor_spl::token_2022::burn#378a
-jne r0, 2, +0x320
-ldxdw r7, [r10-480]
-stxdw [r7+440], r8
-ldxdw r1, [r7+88]
+jne r0, 2, +0x1060
+ldxdw r3, [r10-376]
+stxdw [r3+440], r7
+ldxdw r1, [r3+88]
 ldxdw r1, [r1]
 ldxdw r2, [r1+24]
+stxdw [r10-88], r2
+ldxdw r2, [r1+16]
 stxdw [r10-96], r2
+ldxdw r2, [r1+8]
+stxdw [r10-104], r2
+ldxdw r1, [r1]
+stxdw [r10-112], r1
+ldxdw r1, [r3+344]
+ldxdw r1, [r1]
+ldxdw r2, [r1+24]
+stxdw [r10-56], r2
 ldxdw r2, [r1+16]
-stxdw [r10-104], r2
+stxdw [r10-64], r2
 ldxdw r2, [r1+8]
-stxdw [r10-112], r2
-ldxdw r1, [r1]
-stxdw [r10-120], r1
-ldxdw r1, [r7+344]
-ldxdw r1, [r1]
-ldxdw r2, [r1+24]
-stxdw [r10-64], r2
-ldxdw r2, [r1+16]
 stxdw [r10-72], r2
-ldxdw r2, [r1+8]
-stxdw [r10-80], r2
-ldxdw r1, [r1]
-stxdw [r10-88], r1
-mov64 r1, r10
-add64 r1, -56
+ldxdw r1, [r1]
+stxdw [r10-80], r1
+mov64 r1, r10
+add64 r1, -48
 call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
-ldxw r1, [r10-56]
-jeq r1, 1, +0x15b8
-ldxdw r1, [r10-120]
-stxdw [r10-336], r1
+ldxw r1, [r10-48]
+jeq r1, 1, +0x1620
 ldxdw r1, [r10-112]
 stxdw [r10-328], r1
 ldxdw r1, [r10-104]
@@ -305,7 +219,9 @@
 stxdw [r10-288], r1
 ldxdw r1, [r10-64]
 stxdw [r10-280], r1
-ldxdw r6, [r10-16]
+ldxdw r1, [r10-56]
+stxdw [r10-272], r1
+ldxdw r6, [r10-8]
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 lddw r1, 0x300000000
 ldxdw r2, [r1]
@@ -313,35 +229,35 @@
 mov64 r4, 1
 mov64 r1, r2
 add64 r1, -256
-jle r1, r2, +0xa80
-jeq r4, 0, +0xa90
+jle r1, r2, +0x790
+jeq r4, 0, +0x7a0
 lddw r1, 0x300007f00
-jne r2, 0, +0xab0
+jne r2, 0, +0x7c0
 lddw r2, 0x300000007
-jgt r1, r2, +0xad0
-ja +0xff0
+jgt r1, r2, +0x7e0
+ja +0xd18
 mov64 r4, 0
-jne r4, 0, +0xa48
+jne r4, 0, +0x758
 mov64 r3, r1
 lddw r1, 0x300007f00
-jeq r2, 0, +0xa60
+jeq r2, 0, +0x770
 mov64 r1, r3
 lddw r2, 0x300000007
-jle r1, r2, +0xff0
+jle r1, r2, +0xd18
 mov64 r9, r10
-add64 r9, -304
+add64 r9, -296
 lddw r2, 0x300000000
 stxdw [r2], r1
-stxdw [r1+8], r8
+stxdw [r1+8], r7
 lddw r2, 0xfaee7c52752f5921
 stxdw [r1], r2
-ldxdw r2, [r10-336]
+ldxdw r2, [r10-328]
 stxdw [r1+16], r2
-ldxdw r2, [r10-328]
+ldxdw r2, [r10-320]
 stxdw [r1+24], r2
-ldxdw r2, [r10-320]
+ldxdw r2, [r10-312]
 stxdw [r1+32], r2
-ldxdw r2, [r10-312]
+ldxdw r2, [r10-304]
 stxdw [r1+40], r2
 ldxdw r2, [r9]
 stxdw [r1+48], r2
@@ -352,69 +268,68 @@
 ldxdw r2, [r9+24]
 stxdw [r1+72], r2
 stxdw [r1+80], r6
-stxdw [r10-56], r1
-stdw [r10-48], 88
-mov64 r1, r10
-add64 r1, -56
+stxdw [r10-48], r1
+stdw [r10-40], 88
+mov64 r1, r10
+add64 r1, -48
 mov64 r6, 1
 mov64 r2, 1
 call anchor_lang::solana_program::log::sol_log_data#9398
-ldxdw r1, [r7+512]
-ldxdw r4, [r1+8]
-ldxdw r2, [r4]
+ldxdw r1, [r10-376]
+ldxdw r1, [r1+512]
+ldxdw r7, [r1+8]
+ldxdw r2, [r7]
 add64 r2, 1
-jeq r2, 0, +0xc08
+jeq r2, 0, +0x920
 mov64 r6, 0
 ldxdw r8, [r1]
-stxdw [r4], r2
-jeq r6, 1, +0x1500
-ldxdw r5, [r1+16]
-ldxdw r2, [r5]
+stxdw [r7], r2
+jeq r6, 1, +0x15b8
+ldxdw r6, [r1+16]
+ldxdw r2, [r6]
 add64 r2, 1
 mov64 r3, 1
-jeq r2, 0, +0xc50
+jeq r2, 0, +0x968
 mov64 r3, 0
-stxdw [r10-352], r4
-stxdw [r5], r2
-jeq r3, 1, +0x1500
-stxdw [r10-360], r5
+stxdw [r6], r2
+jeq r3, 1, +0x15b8
 ldxb r2, [r1+42]
+stxdw [r10-424], r2
+ldxb r2, [r1+41]
+stxdw [r10-416], r2
+ldxb r2, [r1+40]
+stxdw [r10-408], r2
+ldxdw r2, [r1+32]
 stxdw [r10-400], r2
-ldxb r2, [r1+41]
-stxdw [r10-392], r2
-ldxb r2, [r1+40]
-stxdw [r10-384], r2
-ldxdw r2, [r1+32]
-stxdw [r10-376], r2
 ldxdw r1, [r1+24]
-stxdw [r10-368], r1
-ldxdw r1, [r7+88]
+stxdw [r10-392], r1
+ldxdw r3, [r10-376]
+ldxdw r1, [r3+88]
 ldxdw r1, [r1]
 ldxdw r2, [r1+24]
+stxdw [r10-88], r2
+ldxdw r2, [r1+16]
 stxdw [r10-96], r2
+ldxdw r2, [r1+8]
+stxdw [r10-104], r2
+ldxdw r1, [r1]
+stxdw [r10-112], r1
+ldxdw r1, [r3+344]
+ldxdw r1, [r1]
+ldxdw r2, [r1+24]
+stxdw [r10-56], r2
 ldxdw r2, [r1+16]
-stxdw [r10-104], r2
+stxdw [r10-64], r2
 ldxdw r2, [r1+8]
-stxdw [r10-112], r2
-ldxdw r1, [r1]
-stxdw [r10-120], r1
-ldxdw r1, [r7+344]
-ldxdw r1, [r1]
-ldxdw r2, [r1+24]
-stxdw [r10-64], r2
-ldxdw r2, [r1+16]
 stxdw [r10-72], r2
-ldxdw r2, [r1+8]
-stxdw [r10-80], r2
-ldxdw r1, [r1]
-stxdw [r10-88], r1
-mov64 r1, r10
-add64 r1, -56
+ldxdw r1, [r1]
+stxdw [r10-80], r1
+mov64 r1, r10
+add64 r1, -48
 call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
-ldxw r1, [r10-56]
-jeq r1, 1, +0x1618
-ldxdw r1, [r10-120]
-stxdw [r10-336], r1
+ldxw r1, [r10-48]
+jeq r1, 1, +0x1680
+stxdw [r10-448], r6
 ldxdw r1, [r10-112]
 stxdw [r10-328], r1
 ldxdw r1, [r10-104]
@@ -429,7 +344,9 @@
 stxdw [r10-288], r1
 ldxdw r1, [r10-64]
 stxdw [r10-280], r1
-ldxdw r6, [r10-16]
+ldxdw r1, [r10-56]
+stxdw [r10-272], r1
+ldxdw r6, [r10-8]
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 lddw r1, 0x300000000
 ldxdw r1, [r1]
@@ -437,26 +354,27 @@
 mov64 r4, 1
 mov64 r3, r1
 add64 r3, -256
-jle r3, r1, +0xf98
-jeq r4, 0, +0xfa8
+jle r3, r1, +0xcb8
+stxdw [r10-432], r7
+jeq r4, 0, +0xcd0
 lddw r7, 0x300007f00
-jne r1, 0, +0xfc8
-stxdw [r10-408], r8
+jne r1, 0, +0xcf0
+stxdw [r10-440], r8
 lddw r1, 0x300000007
-jle r7, r1, +0xff0
+jle r7, r1, +0xd18
 lddw r8, 0x300000000
 stxdw [r8], r7
-ldxdw r1, [r10-344]
+ldxdw r1, [r10-384]
 stxdw [r7+8], r1
 lddw r1, 0xfaee7c52752f5921
 stxdw [r7], r1
-ldxdw r1, [r10-336]
+ldxdw r1, [r10-328]
 stxdw [r7+16], r1
-ldxdw r1, [r10-328]
+ldxdw r1, [r10-320]
 stxdw [r7+24], r1
-ldxdw r1, [r10-320]
+ldxdw r1, [r10-312]
 stxdw [r7+32], r1
-ldxdw r1, [r10-312]
+ldxdw r1, [r10-304]
 stxdw [r7+40], r1
 ldxdw r1, [r9]
 stxdw [r7+48], r1
@@ -469,43 +387,126 @@
 stxdw [r7+80], r6
 mov64 r1, 0
 mov64 r6, 96
-jsge r6, 0, +0x1018
+jsge r6, 0, +0xe80
 mov64 r2, 96
 lddw r3, panic_location(/home/runner/work/platform-tools/platform-tools/out/rust/library/core/src/iter/traits/iterator.rs:2027:9)
 call alloc::raw_vec::handle_error#4993
 mov64 r4, 0
-jne r4, 0, +0xe60
+stxdw [r10-432], r7
+jne r4, 0, +0xb80
 mov64 r2, r3
 lddw r7, 0x300007f00
-jeq r1, 0, +0xe78
+jeq r1, 0, +0xb98
 mov64 r7, r2
-stxdw [r10-408], r8
+stxdw [r10-440], r8
 lddw r1, 0x300000007
-jgt r7, r1, +0xe98
+jgt r7, r1, +0xbb8
 mov64 r1, 1
 mov64 r2, 256
-lddw r3, panic_location(programs/mayhem-program/src/instructions/burn.rs:10:1)
+lddw r3, panic_location(programs/mayhem-program/src/instructions/burn.rs:67:1)
 call alloc::raw_vec::handle_error#4993
+call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
+lddw r1, 0x300000000
+ldxdw r1, [r1]
+mov64 r3, r1
+add64 r3, -18
+mov64 r2, 0
+mov64 r4, 1
+jle r3, r1, +0x1068
+jeq r4, 0, +0x1078
+lddw r6, 0x300007fee
+jne r1, 0, +0x1098
+lddw r1, 0x300000007
+jle r6, r1, +0x10b8
+lddw r1, 0x300000000
+stxdw [r1], r6
+lddw r1, 0x646e45746f4e6564
+stxdw [r6+8], r1
+lddw r1, 0x6f4d6d656879614d
+stxdw [r6], r1
+sth [r6+16], 25701
+mov64 r1, r10
+add64 r1, -272
+mov64 r2, 5
+call <T as alloc::string::SpecToString>::spec_to_string#5680
+stxdw [r10-288], r6
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-320], r1
+stw [r10-176], 6005
+stdw [r10-280], 18
+stdw [r10-296], 18
+stb [r10-248], 2
+stw [r10-304], 40
+ja +0x1038
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 ldxdw r1, [r8]
 lddw r3, 0x300008000
-jeq r1, 0, +0x1048
+jeq r1, 0, +0xeb0
 mov64 r3, r1
 mov64 r2, r3
 sub64 r2, r6
 mov64 r6, 0
 mov64 r1, 1
 mov64 r4, 1
-jle r2, r3, +0x10a0
-jeq r4, 0, +0x10b0
+jle r2, r3, +0x10e0
+jeq r4, 0, +0x10f0
 lddw r2, 0x300000007
-jgt r6, r2, +0x10d0
-ja +0xf78
+jgt r6, r2, +0x1110
+ja +0xc98
+call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
+lddw r1, 0x300000000
+ldxdw r1, [r1]
+mov64 r3, r1
+add64 r3, -15
+mov64 r2, 0
+mov64 r4, 1
+jle r3, r1, +0x1540
+jeq r4, 0, +0x1550
+lddw r6, 0x300007ff1
+jne r1, 0, +0x1570
+lddw r1, 0x300000007
+jle r6, r1, +0x1590
+lddw r1, 0x300000000
+stxdw [r1], r6
+lddw r1, 0x6465646e4565646f
+stxdw [r6+7], r1
+lddw r1, 0x6f4d6d656879614d
+stxdw [r6], r1
+mov64 r1, r10
+add64 r1, -272
+mov64 r2, 4
+call <T as alloc::string::SpecToString>::spec_to_string#5680
+stxdw [r10-288], r6
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-320], r1
+stw [r10-176], 6004
+stdw [r10-280], 15
+stdw [r10-296], 15
+stb [r10-248], 2
+stw [r10-304], 43
+stdw [r10-312], 48
+stdw [r10-328], 0
+mov64 r1, r10
+add64 r1, -328
+call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
+exit
 mov64 r4, 0
-jne r4, 0, +0x1080
+jne r4, 0, +0xd90
+mov64 r2, r3
+lddw r6, 0x300007fee
+jeq r1, 0, +0xda8
+mov64 r6, r2
+lddw r1, 0x300000007
+jgt r6, r1, +0xdc0
+mov64 r1, 1
+mov64 r2, 18
+lddw r3, panic_location(/home/runner/work/platform-tools/platform-tools/out/rust/library/alloc/src/slice.rs:449:29)
+call alloc::raw_vec::handle_error#4993
+mov64 r4, 0
+jne r4, 0, +0xee8
 mov64 r6, r2
 lddw r2, 0x300000007
-jle r6, r2, +0xf78
+jle r6, r2, +0xc98
 lddw r1, 0x300000000
 stxdw [r1], r6
 lddw r1, 0x1d9acb512ea545e4
@@ -518,7 +519,7 @@
 ldxb r3, [r3]
 stxb [r2+8], r3
 add64 r1, 1
-jne r1, 88, +0x1108
+jne r1, 88, +0x1148
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 lddw r1, 0x300000000
 ldxdw r2, [r1]
@@ -526,55 +527,54 @@
 add64 r1, -34
 mov64 r3, 0
 mov64 r4, 1
-ldxdw r8, [r10-480]
-jle r1, r2, +0x1398
-jeq r4, 0, +0x13a8
+jle r1, r2, +0x13c8
+ldxdw r7, [r10-432]
+jeq r4, 0, +0x13e0
 lddw r1, 0x300007fde
-jne r2, 0, +0x13c8
+jne r2, 0, +0x1400
 lddw r2, 0x300000008
-jlt r1, r2, +0x13e8
+jlt r1, r2, +0x1420
 lddw r2, 0x300000000
 stxdw [r2], r1
-ldxdw r7, [r10-408]
-ldxdw r2, [r7]
-ldxdw r3, [r7+8]
-ldxdw r4, [r7+16]
-ldxdw r5, [r7+24]
+ldxdw r9, [r10-440]
+ldxdw r2, [r9]
+ldxdw r3, [r9+8]
+ldxdw r4, [r9+16]
+ldxdw r5, [r9+24]
 stxdw [r1+24], r5
 stxdw [r1+16], r4
 stxdw [r1+8], r3
 stxdw [r1], r2
 sth [r1+32], 1
-stxdw [r10-48], r1
-stdw [r10-40], 1
-stdw [r10-56], 1
-mov64 r9, r10
-add64 r9, -336
+stxdw [r10-40], r1
+stdw [r10-32], 1
+stdw [r10-48], 1
+mov64 r8, r10
+add64 r8, -328
 mov64 r5, r10
-add64 r5, -56
-mov64 r1, r9
+add64 r5, -48
+mov64 r1, r8
 lddw r2, .rodata[052ae5d7a7daa724]
 mov64 r3, r6
 mov64 r4, 96
 call solana_instruction::Instruction::new_with_bytes#33e2
+ldxdw r1, [r10-424]
+stxb [r10-6], r1
+ldxdw r1, [r10-416]
+stxb [r10-7], r1
+ldxdw r1, [r10-408]
+stxb [r10-8], r1
 ldxdw r1, [r10-400]
-stxb [r10-14], r1
+stxdw [r10-16], r1
 ldxdw r1, [r10-392]
-stxb [r10-15], r1
-ldxdw r1, [r10-384]
-stxb [r10-16], r1
-ldxdw r1, [r10-376]
 stxdw [r10-24], r1
-ldxdw r1, [r10-368]
+ldxdw r1, [r10-448]
 stxdw [r10-32], r1
-ldxdw r1, [r10-360]
-stxdw [r10-40], r1
-ldxdw r1, [r10-352]
-stxdw [r10-48], r1
-stxdw [r10-56], r7
+stxdw [r10-40], r7
+stxdw [r10-48], r9
 mov64 r2, r10
-add64 r2, -56
-mov64 r1, r9
+add64 r2, -48
+mov64 r1, r8
 mov64 r3, 1
 lddw r4, .data.rel.ro+obj
 mov64 r5, 1
@@ -583,85 +583,91 @@
 mov64 r1, r0
 lsh64 r1, 32
 rsh64 r1, 32
-jne r1, 26, +0x1400
+jne r1, 26, +0x1438
 mov64 r0, 2
-ldxdw r3, [r10-352]
-ja +0x1418
+ja +0x1450
 mov64 r4, 0
-jne r4, 0, +0x11a0
+ldxdw r7, [r10-432]
+jne r4, 0, +0x11e0
 mov64 r3, r1
 lddw r1, 0x300007fde
-jeq r2, 0, +0x11b8
+jeq r2, 0, +0x11f8
 mov64 r1, r3
 lddw r2, 0x300000008
-jge r1, r2, +0x11d0
+jge r1, r2, +0x1210
 mov64 r1, 1
 mov64 r2, 34
 call alloc::alloc::handle_alloc_error#decd
 mov64 r1, r0
 call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
-ldxdw r3, [r10-48]
-ldxdw r2, [r3]
+ldxdw r7, [r10-40]
+ldxdw r2, [r7]
 add64 r2, -1
-stxdw [r3], r2
-jne r2, 0, +0x1478
+stxdw [r7], r2
+jne r2, 0, +0x14b0
 mov64 r2, r10
-add64 r2, -48
+add64 r2, -40
 mov64 r6, r1
 mov64 r1, r2
 mov64 r7, r0
 call alloc::rc::Rc<T,A>::drop_slow#7561
 mov64 r1, r6
 mov64 r0, r7
-ldxdw r2, [r10-40]
+ldxdw r2, [r10-32]
 ldxdw r3, [r2]
 add64 r3, -1
 stxdw [r2], r3
-jne r3, 0, +0x14e0
+jne r3, 0, +0x1518
 mov64 r2, r10
-add64 r2, -40
+add64 r2, -32
 mov64 r6, r1
 mov64 r1, r2
 mov64 r7, r0
 call alloc::rc::Rc<T,A>::drop_slow#d793
 mov64 r1, r6
 mov64 r0, r7
-jne r0, 2, +0x320
-stb [r8+448], 2
+jne r0, 2, +0x1060
+ldxdw r1, [r10-376]
+stb [r1+448], 2
 mov64 r0, 2
-ja +0x320
+ja +0x1060
+mov64 r4, 0
+jne r4, 0, +0xf58
+mov64 r2, r3
+lddw r6, 0x300007ff1
+jeq r1, 0, +0xf70
+mov64 r6, r2
+lddw r1, 0x300000007
+jgt r6, r1, +0xf88
+mov64 r1, 1
+mov64 r2, 15
+lddw r3, panic_location(/home/runner/work/platform-tools/platform-tools/out/rust/library/alloc/src/slice.rs:449:29)
+call alloc::raw_vec::handle_error#4993
 call abort
-ldxdw r1, [r10-332]
-stxdw [r10-56], r1
+ldxdw r1, [r10-324]
+stxdw [r10-48], r1
 mov64 r3, r10
-add64 r3, -56
+add64 r3, -48
 lddw r1, str("called `Result::unwrap()` on an `Err` value")
 mov64 r2, 43
 lddw r4, table[3:<solana_program_error::ProgramError as core::fmt::Debug>::fmt#4fcd]
-lddw r5, panic_location(programs/mayhem-program/src/instructions/burn.rs:77:52)
+lddw r5, panic_location(programs/mayhem-program/src/instructions/burn.rs:40:70)
 call core::result::unwrap_failed#5a05
+ldxdw r1, [r10-44]
+stxdw [r10-328], r1
 mov64 r3, r10
-add64 r3, -1
-lddw r1, str("a Display implementation returned an error unexpectedly")
-mov64 r2, 55
-lddw r4, table[3:<core::fmt::Error as core::fmt::Debug>::fmt#aaa5]
-lddw r5, panic_location(/home/runner/work/platform-tools/platform-tools/out/rust/library/alloc/src/string.rs:2830:14)
-call core::result::unwrap_failed#5a05
-ldxdw r1, [r10-52]
-stxdw [r10-336], r1
-mov64 r3, r10
-add64 r3, -336
+add64 r3, -328
 lddw r1, str("called `Result::unwrap()` on an `Err` value")
 mov64 r2, 43
 lddw r4, table[3:<solana_program_error::ProgramError as core::fmt::Debug>::fmt#4fcd]
-lddw r5, panic_location(programs/mayhem-program/src/instructions/burn.rs:104:33)
+lddw r5, panic_location(programs/mayhem-program/src/instructions/burn.rs:59:33)
 call core::result::unwrap_failed#5a05
-ldxdw r1, [r10-52]
-stxdw [r10-336], r1
+ldxdw r1, [r10-44]
+stxdw [r10-328], r1
 mov64 r3, r10
-add64 r3, -336
+add64 r3, -328
 lddw r1, str("called `Result::unwrap()` on an `Err` value")
 mov64 r2, 43
 lddw r4, table[3:<solana_program_error::ProgramError as core::fmt::Debug>::fmt#4fcd]
-lddw r5, panic_location(programs/mayhem-program/src/instructions/burn.rs:110:33)
+lddw r5, panic_location(programs/mayhem-program/src/instructions/burn.rs:62:33)
 call core::result::unwrap_failed#5a05
